### PR TITLE
Merge 3.x

### PR DESCRIPTION
**Pull request by core23**
Manual merge of 3.x into the master branch.

Refs sonata-project/dev-kit#221

### CODE REVIEW

File: Tests/Controller/Api/MediaControllerTest.php [hunk 1]
```diff
@@ -31,11 +31,13 @@ public function testGetMediaAction()
...
         $mController = $this->createMediaController($mManager);
 
-        $params = $this->getMock('FOS\RestBundle\Request\ParamFetcherInterface');
-        $params->expects($this->once())->method('all')->will($this->returnValue(array('page' => 1, 'count' => 10, 'orderBy' => array('id' => 'ASC'))));
-        $params->expects($this->exactly(3))->method('get');
+        $paramFetcher = $this->getMockBuilder('FOS\RestBundle\Request\ParamFetcher')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $paramFetcher->expects($this->exactly(3))->method('get');
+        $paramFetcher->expects($this->once())->method('all')->will($this->returnValue(array()));
```
Comment (greg0ire) [inline Tests/Controller/Api/MediaControllerTest.php]: @core23 : a very similar change is missing in the Gallery controller test. It was introduced in #1155

**Comment by core23** [comment]
Started a new clean merge...

**Comment by greg0ire** [comment]
There are many many commits to merge. You can try merging half of them, then another half etc… etc…, it might be easier. Just use `git merge commit hash` (never tried that, but it should work).

**Comment by core23** [comment]
Next try... until tag 3.2.0

**Comment by greg0ire** [comment]
Doesn't look too good :( … maybe try until 3.1.0 ?

**Comment by greg0ire** [comment]
I've tried the build locally, without any merge, and it fails, although with other error messages :(

**Comment by core23** [comment]
I give up. Maybe you can give try @greg0ire ?

**Comment by greg0ire** [comment]
Ok I will. I'll first check the master build with Travis.

**Comment by greg0ire** [comment]
See #1162

**Comment by greg0ire** [comment]
See #1163

**Pull request closed without merging**